pageserver/src/walredo.rs

```diff
@@ -238,23 +244,69 @@ impl PostgresRedoManager {
         pg_version: u32,
     ) -> Result<Bytes, WalRedoError> {
         let (rel, blknum) = key_to_rel_block(key).or(Err(WalRedoError::InvalidRecord))?;
+        // Relational WAL records are applied using wal-redo-postgres
+        let buf_tag = BufferTag { rel, blknum };
 
         let start_time = Instant::now();
 
         let mut process_guard = self.process.lock().unwrap();
+
         let lock_time = Instant::now();
 
         // launch the WAL redo process on first use
         if process_guard.is_none() {
             let p = PostgresRedoProcess::launch(self.conf, self.tenant_id, pg_version)?;
             *process_guard = Some(p);
         }
+
+        if let Some(pipe) = self.pipe.as_ref() {
+            drop(process_guard);
+            let tag_len = 1 + 4 * 4;
+            let ch = 1;
+            let len = 4;
+
+            let mut buf = Vec::with_capacity(
+                (ch + len + tag_len)
```

Comment:
you reserve `ch + len + tag_len` twice (look at the bottom). Is it by design? if so, just multiply it

Reply:
I usually try to keep these as explicit as possible for easier code review, leave const folding to the compiler.